module/product/src/Infrastructure/Grid/Builder/Query/ProductRelationAttributeDataSetQueryBuilder.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * Copyright © Ergonode Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Product\Infrastructure\Grid\Builder\Query;
+
+use Ergonode\Attribute\Domain\Entity\AbstractAttribute;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Ergonode\Core\Domain\ValueObject\Language;
+use Ergonode\Product\Domain\Entity\Attribute\ProductRelationAttribute;
+
+class ProductRelationAttributeDataSetQueryBuilder extends AbstractAttributeDataSetBuilder
+{
+    public function supports(AbstractAttribute $attribute): bool
+    {
+        return $attribute instanceof ProductRelationAttribute;
+    }
+
+    public function addSelect(QueryBuilder $query, string $key, AbstractAttribute $attribute, Language $language): void
+    {
+        $info = $this->query->getLanguageNodeInfo($this->resolver->resolve($attribute, $language));
+
+        $sql = sprintf(
+            '(SELECT 	
+			            DISTINCT ON (product_id) product_id, 
```

Comment:
Could you format the query appoprietly? Currently, I think indentations do not make sense.